test/tests/meshgenerators/xyz_delaunay_generator/tests

```diff
@@ -0,0 +1,55 @@
+[Tests]
+  design = 'meshgenerators/XYZDelaunayGenerator.md'
+  issues = '#28297'
+  [xyzdelaunay]
+    requirement = "The system shall have the capability of tetrahedralizing a triangulated boundary"
+    [basic]
+      type = 'CSVDiff'
+      input = 'xyzdelaunay_mesh_generator.i'
+      csvdiff = 'xyzdelaunay_mesh_generator_out.csv'
+      recover = false
+      detail = "within an input boundary mesh"
+    []
+    [with_holes]
+      type = 'CSVDiff'
+      input = 'xyzdelaunay_with_holes.i'
+      csvdiff = 'xyzdelaunay_with_holes_out.csv'
+      recover = false
+      detail = "respecting any specified interior 'hole' meshes"
+    []
+    [stitching]
+      type = 'CSVDiff'
+      input = 'xyzdelaunay_stitching.i'
+      cli_args = 'Mesh/final_generator=triang'
+      csvdiff = 'xyzdelaunay_stitching_out.csv'
+      recover = false
+      detail = "selectively stitching 'hole' meshes into the final mesh"
+    []
+    [nested]
+      type = 'CSVDiff'
+      input = 'xyzdelaunay_nested.i'
+      csvdiff = 'xyzdelaunay_nested_out.csv'
+      recover = false
+      detail = "allowing deep nesting of tetrahedralizations within tetrahedralizations"
+    []
+    [smoothed]
+      type = 'CSVDiff'
+      input = 'xyzdelaunay_smoothed.i'
+      csvdiff = 'xyzdelaunay_smoothed_out.csv'
+      recover = false
+      detail = "with optional Laplacian mesh smoothing."
+    []
+  []
+  [errors]
+    requirement = "The system shall throw an error when tetrahedralizing a triangulated boundary"
+    [err_conflicting_name_map]
+      type = RunException
+      input = 'xyzdelaunay_stitching.i'
+      cli_args = '--mesh-only err.e
+                  Mesh/triang/stitch_holes="true true"
+                  Mesh/triang/holes="hole_1_name hole_2_name_2"'
+      expect_err = "The hole meshes contain subdomain name maps with conflicts."
+      detail = "if the input hole meshes contain subdomain name maps with conflicts"
+    []
+  []
```

Comment:
should it support an open domain? With a boundary that is not closed

Reply:
No, and it's got a check on the libMesh side for that.  Looking at that again, though ... any suggestions for improvements on the error message?

```
       "Missing element when trying to identify boundary component to mesh.\n"
       "Perhaps a boundary element does not have all neighbor pointers set?\n"
```

That "does not have all neighbor pointers set?" suggestion is probably more of a red herring at this point, since I stopped assuming that input meshes were prepared and started fixing up neighbor pointers myself.

I'm thinking something like:

```
       "Tet generation encountered a 2D element with a null neighbor, but a\n"
       "boundary surface must be a closed 2D manifold.\n"
```

Reply:
is manifold too mathy for users?
could do 2D closed manifold (surface) ?

"with a null neighbor outside of the prescribed holes and boundaries"

Reply:
I'll add "(surface)".

"outside of the prescribed holes and boundaries" isn't actually correct, though - both the outer boundary and each hole boundary should not have any boundaries.  ∂∘∂=0 for the boundary operator on manifolds-with-boundary, in mathy parlance.  ;-)